### PR TITLE
fix: fails to massage pnpm release notes

### LINKED ISSUE #16921: Fails to massage pnpm release notes
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Please select which platform you are using if self-hosting.

_No response_

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

Renovate fails to massage issue links in the `What's Changed` section of pnpm release notes, leading to autolinks, e.g., https://github.com/renovatebot/docker-buildpack/pull/338.

It seems to be caused by an HTML table; the raw release note is as follows:

### Relevant debug logs

<details><summary>Raw Release Note</summary>

```markdown
## Minor Changes

-   A new setting supported: `prefer-symlinked-executables`. When `true`, pnpm will create symlinks to executables in
    `node_modules/.bin` instead of command shims (but on POSIX systems only).

    This setting is `true` by default when `node-linker` is set to `hoisted`.

    Related issue: [#4782](https://github.com/pnpm/pnpm/issues/4782).

-   When `lockfile-include-tarball-url` is set to `true`, every entry in `pnpm-lock.yaml` will contain the full URL to the package's tarball [#5054](https://github.com/pnpm/pnpm/pull/5054).

## Patch Changes

-   `pnpm deploy` should include all dependencies by default [#5035](https://github.com/pnpm/pnpm/issues/5035).
-   Don't print warnings about file verifications. Just print info messages instead.
-   `pnpm publish --help` should print the `--recursive` and `--filter` options [#5019](https://github.com/pnpm/pnpm/issues/5019).
-   It should be possible to run exec/run/dlx with the `--use-node-version` option.
-   `pnpm deploy` should not modify the lockfile [#5071](https://github.com/pnpm/pnpm/issues/5071)
-   `pnpm deploy` should not fail in CI [#5071](https://github.com/pnpm/pnpm/issues/5071)
-   When `auto-install-peers` is set to `true`, automatically install direct peer dependencies [#5028](https://github.com/pnpm/pnpm/pull/5067).

    So if your project the next manifest:

    ```json
    {
      "dependencies": {
        "lodash": "^4.17.21"
      },
      "peerDependencies": {
        "react": "^18.2.0"
      }
    }
    ```

    pnpm will install both lodash and react as a regular dependencies.

## Our Gold Sponsors

<table>
  <tbody>
    <tr>
      <td align="center" valign="middle">
        <a href="https://bit.dev/?utm_source=pnpm&utm_medium=release_notes" target="_blank"></a>
      </td>
      <td align="center" valign="middle">
        <a href="https://nhost.io/?utm_source=pnpm&utm_medium=release_notes" target="_blank"></a>
      </td>
      <td align="center" valign="middle">
        <a href="https://novu.co/?utm_source=pnpm&utm_medium=release_notes" target="_blank"></a>
      </td>
    </tr>
  </tbody>
</table>

## Our Silver Sponsors

<table>
  <tbody>
    <tr>
      <td align="center" valign="middle">
        <a href="https://prisma.io/?utm_source=pnpm&utm_medium=release_notes" target="_blank"></a>
      </td>
      <td align="center" valign="middle">
        <a href="https://leniolabs.com/?utm_source=pnpm&utm_medium=release_notes" target="_blank"></a>
      </td>
      <td align="center" valign="middle">
        <a href="https://vercel.com/?utm_source=pnpm&utm_medium=release_notes" target="_blank"></a>
      </td>
      <td align="center" valign="middle">
        <a href="https://www.takeshape.io/?utm_source=pnpm&utm_medium=release_notes" target="_blank"></a>
      </td>
    </tr>
  </tbody>
</table>

## What's Changed
* pnpm rebuild accepts --store-dir by @chengcyber in https://github.com/pnpm/pnpm/pull/5036
* fix(deploy): include all deps by default by @zkochan in https://github.com/pnpm/pnpm/pull/5040
* chore(deps): upgrade nock to v13 by @mcmxcdev in https://github.com/pnpm/pnpm/pull/5043
* fix: log more info on HTTP error by @zkochan in https://github.com/pnpm/pnpm/pull/4917
* fix: document the -r option by @zkochan in https://github.com/pnpm/pnpm/pull/5044
* chore(deps): upgrade sinon to v14 by @mcmxcdev in https://github.com/pnpm/pnpm/pull/5045
* fix(audit): add authentication to pnpm-audit by @sled in https://github.com/pnpm/pnpm/pull/5053
* feat: prefer-symlinked-executables by @zkochan in https://github.com/pnpm/pnpm/pull/5048
* chore: update pnpm-workspace.yaml by @ayu14214 in https://github.com/pnpm/pnpm/pull/5060
* feat: add `lockfile-include-tarball-url` option by @MBelniak in https://github.com/pnpm/pnpm/pull/5054
* fix: auto install root peer deps when auto-install-peers=true by @zkochan in https://github.com/pnpm/pnpm/pull/5067
* fix(deploy): don't modify the lockfile and fail in CI by @zkochan in https://github.com/pnpm/pnpm/pull/5074

## New Contributors
* @mcmxcdev made their first contribution in https://github.com/pnpm/pnpm/pull/5043
* @sled made their first contribution in https://github.com/pnpm/pnpm/pull/5053
* @ayu14214 made their first contribution in https://github.com/pnpm/pnpm/pull/5060
* @MBelniak made their first contribution in https://github.com/pnpm/pnpm/pull/5054

**Full Changelog**: https://github.com/pnpm/pnpm/compare/v7.5.2...v7.6.0
```

</details>


### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

### PR DESCRIPTION
**Pull request by MaronHatoum**
## Changes
added regex to detect if '\n' is missing in release notes 

## Context
closes https://github.com/renovatebot/renovate/issues/16921

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by nabeelsaabna** [comment]
can you please add tests

**Review by nabeelsaabna — DISMISSED** [review]
looks good to me 👍

File: lib/util/markdown.ts [hunk 1]
```diff
@@ -18,6 +18,10 @@ export function sanitizeMarkdown(markdown: string): string {
   const backTickRe = regEx(/&#x60;([^/]*?)&#x60;/g);
   res = res.replace(backTickRe, '`$1`');
   res = res.replace(regEx(/`#&#8203;(\d+)`/g), '`#$1`');
+  res = res.replace(
+    regEx(/(?<before>[^\n]\n)(?<title>####.*)/g),
```
Comment (viceice) [inline lib/util/markdown.ts:22]: shouldn't we allow any header level?
```suggestion
    regEx(/(?<before>[^\n]\n)(?<title>#.*)/g),
```

**Review by rarkins — CHANGES_REQUESTED** [review]
The test you added passes even if the code you added is removed. In other words, it's not testing that your changes work. Test should be one which fails on `main` but passes for the PR

File: lib/util/__fixtures__/release-notes.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#### Our Silver Sponsors
+
+<table>
+</table>
+
+#### What's Changed
+* pnpm rebuild accepts --store-dir by @&#8203;user in https://github.com/foo/foo/pull/1
+
+#### New Contributors
+* @&#8203;user made their first contribution in https://github.com/foo/foo/pull/2
```
Comment (rarkins) [inline lib/util/__fixtures__/release-notes.txt:15]: Is there a reason why y0ou're using `@&#8203;`? It's a trick we use, but I've never seen it in raw release notes

Comment (MaronHatoum) [reply lib/util/__fixtures__/release-notes.txt:15]: I ran Renovate on the reproduction repository and this is part of the returned value from `sanitizeMarkdown` function

image

**Review by rarkins — APPROVED** [review]

**Pull request merged**